src/axolotl/utils/callbacks.py

```diff
@@ -234,32 +241,68 @@ def on_evaluate(
                 )
                 # There are two tokens, the output, and eos token.
                 for i, logit in enumerate(logits):
-                    label_non_zero_id = (batch["labels"][i] != -100).nonzero()[0][0]
+                    label_non_zero_id = (batch["labels"][i] != IGNORE_INDEX).nonzero()[
+                        0
+                    ][0]
                     logit_abcd = logit[label_non_zero_id - 1][abcd_idx]
                     preds.append(torch.argmax(logit_abcd).item())
                 labels = labels[labels != IGNORE_INDEX].view(-1, 2)[:, 0]
                 refs += [
-                    abcd_idx.index(label) if labels in abcd_idx else -1
+                    abcd_idx.index(label) if label in abcd_idx else -1
```

Comment:
😮

Reply:
haha, yeah, I don't know how that slipped through the first time (or even worked)